### PR TITLE
Convenient transforms to Infallible and back

### PR DESCRIPTION
**Pull request by asiliuk**
Adds convenient `asInfallible()` to types that can't fail. And `asDriver()` to `InfallibleType`
Let me know if there is a better way to implement it than just assert on failure.
Ideal variant would be to do something similar to how `asDriver` is implemented, but there's no access `RxRelay` in `RxSwift` module and there's no access to `Infallible(_ source: Observable<Element>)` in `RxRelay` module

### CODE REVIEW

File: RxCocoa/Traits/ControlEvent.swift [hunk 1]
```diff
@@ -65,4 +65,12 @@ public struct ControlEvent<PropertyType> : ControlEventType {
...
         self
     }
+    
+    /// - returns: `Infallible` interface.
+    public func asInfallible() -> Infallible<Element> {
+        self.asInfallible { error in
+            rxFatalErrorInDebug("Received unexpected error event. \(error)")
+            return .empty()
+        }
+    }
```
Comment (freak4pc) [inline RxCocoa/Traits/ControlEvent.swift:75]:
```suggestion
    public func asInfallible() -> Infallible<Element> {
       asInfallible(onErrorFallbackTo: .empty())
    }
```

File: RxCocoa/Traits/Driver/Infallible+Driver.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+extension InfallibleType {
+    /// Converts `InfallibleType` to `Driver`.
+    ///
+    /// - returns: Observable sequence.
+    public func asDriver() -> Driver<Element> {
+        let source = self.asObservable()
+            .observe(on:DriverSharingStrategy.scheduler)
+        return SharedSequence(source)
+    }
```
Comment (freak4pc) [inline RxCocoa/Traits/Driver/Infallible+Driver.swift:19]: I'd go for a single-liner here
```suggestion
    public func asDriver() -> Driver<Element> {
        SharedSequence(asObservable().observe(on: DriverSharingStrategy.scheduler))
    }
```

File: RxCocoa/Traits/SharedSequence/SharedSequence.swift [hunk 3]
```diff
@@ -55,6 +55,14 @@ public struct SharedSequence<SharingStrategy: SharingStrategyProtocol, Element>
     public func asSharedSequence() -> SharedSequence<SharingStrategy, Element> {
         self
     }
+    
+    /// - returns: `Infallible` interface.
+    public func asInfallible() -> Infallible<Element> {
+        self.asInfallible { error in
```
Comment (freak4pc) [inline RxCocoa/Traits/SharedSequence/SharedSequence.swift:61]: like above

File: RxRelay/BehaviorRelay.swift [hunk 4]
```diff
@@ -39,4 +39,14 @@ public final class BehaviorRelay<Element>: ObservableType {
     public func asObservable() -> Observable<Element> {
         self.subject.asObservable()
     }
+    
+    /// Convert to an `Infallible`
+    ///
+    /// - returns: `Infallible<Element>`
+    public func asInfallible() -> Infallible<Element> {
+        self.subject.asInfallible { error in
```
Comment (freak4pc) [inline RxRelay/BehaviorRelay.swift:47]: like above

File: RxRelay/PublishRelay.swift [hunk 5]
```diff
@@ -33,4 +33,14 @@ public final class PublishRelay<Element>: ObservableType {
     public func asObservable() -> Observable<Element> {
         self.subject.asObservable()
     }
+    
+    /// Convert to an `Infallible`
+    ///
+    /// - returns: `Infallible<Element>`
+    public func asInfallible() -> Infallible<Element> {
+        self.subject.asInfallible { error in
```
Comment (freak4pc) [inline RxRelay/PublishRelay.swift:41]: like above

**Review by freak4pc — CHANGES_REQUESTED** [review]
Thank you! Idea seems good to me, just a few notes.

File: RxCocoa/Traits/Driver/Infallible+Driver.swift [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+//
+
+import RxSwift
+
+extension InfallibleType {
+    /// Converts `InfallibleType` to `Driver`.
+    ///
+    /// - returns: Observable sequence.
+    public func asDriver() -> Driver<Element> {
+        SharedSequence(asObservable().observe(on:DriverSharingStrategy.scheduler))
```
Comment (freak4pc) [inline RxCocoa/Traits/Driver/Infallible+Driver.swift:16]:
```suggestion
        SharedSequence(asObservable().observe(on: DriverSharingStrategy.scheduler))
```

**Review by freak4pc — APPROVED** [review]
One note, I'll commit it and merge

**Comment by freak4pc** [comment]
I think you need to run ./scripts/update-jazzy-config.rb (CI failed)

**Comment by asiliuk** [comment]
Thanks for taking your time! Updated jazzy

**Comment by freak4pc** [comment]
Failures seem unrelated. I'll merge, thank you!

**Pull request merged**